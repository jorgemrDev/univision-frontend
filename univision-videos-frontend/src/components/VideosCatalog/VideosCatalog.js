import React, { useState } from "react";
import "./VideosCatalog.scss";
import { Col, Card } from "antd";
import { STATIC_FOLDER_FROM_SERVER, API_URL } from "../../utils/constants";
import { EyeFilled } from "@ant-design/icons";
import ModalVideo from "../../components/ModalVideo";

export default function VideosCatalog({ videos }) {
  console.log(videos);
  return videos.map((video) => (
    <Col key={video.videoId} className="videos-catalog">
      <VideoCard video={video}></VideoCard>
    </Col>
  ));
}

function VideoCard(props) {
  const [isVisibleModal, setIsVisiblemodal] = useState(false);
  const {
    video: { videoId, title, thumbnailPath, videoPath },
  } = props;
  const { Meta } = Card;
  const posterPath = `${API_URL}${STATIC_FOLDER_FROM_SERVER}/${thumbnailPath}`;
  const videoUrl = `${API_URL}${STATIC_FOLDER_FROM_SERVER}/${videoPath}`;

  const openModal = () => setIsVisiblemodal(true);
  const closeModal = () => setIsVisiblemodal(false);

  return (
    <>
      <Card
        onClick={openModal}
        hoverable
        style={{ width: "240px" }}
        cover={<img alt={title} src={posterPath}></img>}
        actions={<EyeFilled />}
      >
        <Meta title={title}></Meta>
      </Card>
      <ModalVideo
        videoPath={videoUrl}
        isOpen={isVisibleModal}
        close={closeModal}
      ></ModalVideo>
    </>
  );
}
