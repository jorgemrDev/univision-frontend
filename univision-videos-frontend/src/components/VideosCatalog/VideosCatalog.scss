.videos-catalog {
  display: flex !important;
  justify-content: center !important;
  margin-top: 25px;
  margin-right: 5px;

  .ant-card-cover {
    img {
      height: 300px !important;
    }
  }
  .ant-card-meta-title {
    text-align: center;
  }
}
