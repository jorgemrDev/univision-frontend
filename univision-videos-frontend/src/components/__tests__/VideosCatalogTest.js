import React from "react";
import VideosCatalog from "../VideosCatalog";
import renderer from "react-test-renderer";

const videos = [
  {
    videoId: "1",
    title: "title",
    thumbnailPath: "thumbnailPath",
    videoPath: "videoPath",
  },
  {
    videoId: "2",
    title: "title2",
    thumbnailPath: "thumbnailPath2",
    videoPath: "videoPath2",
  },
  {
    videoId: "3",
    title: "title3",
    thumbnailPath: "thumbnailPath3",
    videoPath: "videoPath2",
  },
];

test("Render VideosCatalog correctly", () => {
  const component = renderer.create(
    <VideosCatalog videos={videos}></VideosCatalog>
  );
  let tree = component.toJSON();
  expect(tree.length).toBe(3);
  expect(tree).toMatchSnapshot();
});
