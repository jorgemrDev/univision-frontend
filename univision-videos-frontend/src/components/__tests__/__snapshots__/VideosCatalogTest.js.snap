// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render VideosCatalog correctly 1`] = `
Array [
  <div
    className="ant-col videos-catalog"
    style={Object {}}
  >
    <div
      className="ant-card ant-card-bordered ant-card-hoverable"
      onClick={[Function]}
      style={
        Object {
          "width": "240px",
        }
      }
    >
      <div
        className="ant-card-cover"
      >
        <img
          alt="title"
          src="http://localhost:26765/StaticFiles/thumbnailPath"
        />
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              title
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ant-col videos-catalog"
    style={Object {}}
  >
    <div
      className="ant-card ant-card-bordered ant-card-hoverable"
      onClick={[Function]}
      style={
        Object {
          "width": "240px",
        }
      }
    >
      <div
        className="ant-card-cover"
      >
        <img
          alt="title2"
          src="http://localhost:26765/StaticFiles/thumbnailPath2"
        />
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              title2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="ant-col videos-catalog"
    style={Object {}}
  >
    <div
      className="ant-card ant-card-bordered ant-card-hoverable"
      onClick={[Function]}
      style={
        Object {
          "width": "240px",
        }
      }
    >
      <div
        className="ant-card-cover"
      >
        <img
          alt="title3"
          src="http://localhost:26765/StaticFiles/thumbnailPath3"
        />
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <div
          className="ant-card-meta"
        >
          <div
            className="ant-card-meta-detail"
          >
            <div
              className="ant-card-meta-title"
            >
              title3
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
